### ISSUE TITLE
-r (reuse) flag does not seem to be working anymore

### ISSUE THREAD

**Issue by Seefer** [microsoft/vscode#237038]
Type: <b>Bug</b>

I often use VS Code's integrated terminal to open projects whose directory I'm in and use the `code -r .` command to make Code reuse the existing instance. This seems to not work anymore and when I run `code -r .` it always results in another instance of VS Code opening.

This was working fine 4 weeks ago.

VS Code version: Code 1.96.2 (fabdb6a30b49f79a7aba0f2ad9df9b399473380f, 2024-12-19T10:22:47.216Z)
OS version: Windows_NT x64 10.0.22635
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-10900K CPU @ 3.70GHz (20 x 3696)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|63.83GB (52.07GB free)|
|Process Argv|--crash-reporter-id 39706a1d-47ad-4451-8289-40dab0b1bf44|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (38)</summary>

Extension|Author (truncated)|Version
---|---|---
aws-toolkit-vscode|ama|3.40.0
htmltagwrap|bra|1.0.0
vscode-tailwindcss|bra|0.12.17
gitignore|cod|0.9.0
codium|Cod|0.12.7
vscode-eslint|dba|3.0.10
githistory|don|0.6.20
es7-react-js-snippets|dsz|4.4.3
EditorConfig|Edi|0.16.4
prettier-vscode|esb|11.0.0
vscode-firefox-debug|fir|2.12.0
auto-rename-tag|for|0.1.10
codespaces|Git|1.17.3
copilot|Git|1.254.0
copilot-chat|Git|0.23.2
remotehub|Git|0.64.0
vscode-pull-request-github|Git|0.102.0
git-graph|mhu|1.30.0
inline-fold|moa|0.2.6
vscode-docker|ms-|1.29.3
vscode-edge-devtools|ms-|2.1.6
remote-containers|ms-|0.394.0
remote-ssh|ms-|0.116.1
remote-ssh-edit|ms-|0.87.0
remote-wsl|ms-|0.88.5
vscode-remote-extensionpack|ms-|0.26.0
azure-repos|ms-|0.40.0
remote-explorer|ms-|0.4.3
remote-repositories|ms-|0.42.0
remote-server|ms-|1.5.2
typescript-javascript-grammar|ms-|0.0.55
debugger-for-edge|msj|1.0.15
advanced-new-file|pat|1.2.2
LiveServer|rit|5.7.9
gistfs|vsl|0.6.0
quokka-vscode|Wal|1.0.658
wallaby-extension-pack|Wal|1.0.1
wallaby-vscode|Wal|1.0.401

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyonecf:30548226
2i9eh265:30646982
962ge761:30959799
pythonnoceb:30805159
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
dvdeprecation:31068756
dwnewjupyter:31046869
nativerepl1:31139838
pythonrstrctxt:31112756
nativeloc1:31192215
cf971741:31144450
iacca1:31171482
notype1cf:31157160
5fd0e150:31155592
dwcopilot:31170013
stablechunks:31184530
6074i472:31201624

```

</details>

**Comment by Seefer** [comment]
Just realised I could launch VS Code with the --disable-extensions flag. I did this and then tried opening a folder in the integrated terminal using `code -r .` and it worked, leaving me surprised that an extension can possibly override something as fundamental as CLI launch params. 😲.

Time to track down which pesky extension is causing this unwanted behaviour.

**Comment by Seefer** [comment]
Hmmm. Turns out that the **Cursor** IDE fork of VS Code was the culprit and what I was seeing was Cursor being launch using the code command inside integrated terminal. I'd set the Theme to the same as Code so did not immediately notice the slight difference from the IDE UI.

Not cool Cursor. So not cool in fact that I've uninstalled Cursor for seemingly silently overriding the `code` command and annoying me as a result. Cursor devs must try harder at respecting `side-by-side install` concerns.

Oddly, typing `code` from **Microsoft Terminal** does launch VS Code and not Cursor but behaviour changes in VS Code integrated terminal. Maybe some CLI environment issue may need to be looked at?
